samza-core/src/main/scala/org/apache/samza/metrics/reporter/MetricsSnapshotReporterFactory.scala

```diff
@@ -124,7 +124,20 @@ class MetricsSnapshotReporterFactory extends MetricsReporterFactory with Logging
       Util.getTaskClassVersion(config),
       Util.getSamzaVersion,
       Util.getLocalHost.getHostName,
-      serde, blacklist)
+      serde,
+      blacklist,
+      Util.getDeploymentType(config),
+      Util.getApiType(config),
+      Util.getContainerCount(config),
+      Util.getContainerMemoryMb(config),
+      Util.getNumCores(config),
+      Util.getThreadPoolSize(config),
+      Util.getHostAffinityEnabled(config),
+      Util.getSspGrouperFactory(config),
+      Util.getContainerRetryCount(config),
+      Util.getContainerRetryWindowMs(config),
+      Util.getMaxConcurrency(config),
+      Util.getMaxJvmHeapMb)
```

Comment:
Please maintain the existing ordering of passing in the parameters, then the serde and then the blacklist. 
Similar for MetricsHeader above, where time fields are passed in last.